LibreNMS/OS/Ewc.php

```diff
@@ -0,0 +1,108 @@
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2017 James Andrewartha
+ * @author     James Andrewartha <trs80@ucc.asn.au>
+ */
+
+namespace LibreNMS\OS;
+
+use LibreNMS\Device\WirelessSensor;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessApCountDiscovery;
+#use LibreNMS\Interfaces\Discovery\Sensors\WirelessCcqDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessClientsDiscovery;
+#use LibreNMS\Interfaces\Discovery\Sensors\WirelessFrequencyDiscovery;
+#use LibreNMS\Interfaces\Discovery\Sensors\WirelessNoiseFloorDiscovery;
+#use LibreNMS\Interfaces\Discovery\Sensors\WirelessPowerDiscovery;
+#use LibreNMS\Interfaces\Discovery\Sensors\WirelessQualityDiscovery;
+#use LibreNMS\Interfaces\Discovery\Sensors\WirelessRateDiscovery;
+#use LibreNMS\Interfaces\Discovery\Sensors\WirelessRssiDiscovery;
+use LibreNMS\OS;
+
+class Ewc extends OS implements
+    WirelessApCountDiscovery,
+    #WirelessCcqDiscovery,
+    WirelessClientsDiscovery
+    #WirelessFrequencyDiscovery,
+    #WirelessNoiseFloorDiscovery,
+    #WirelessPowerDiscovery,
+    #WirelessQualityDiscovery,
+    #WirelessRateDiscovery,
+    #WirelessRssiDiscovery
+{
+    /**
+     * Returns an array of LibreNMS\Device\Sensor objects
+     *
+     * @return array Sensors
+     */
+    public function discoverWirelessClients()
+    {
+        $apstats =  $this->getCacheByIndex('apStatsMuCounts', "HIPATH-WIRELESS-HWC-MIB");
+        $apnames =  $this->getCacheByIndex('apName', "HIPATH-WIRELESS-HWC-MIB");
+        $sensors = array(
+            new WirelessSensor(
+                'ap-count',
+                $this->getDeviceId(),
+                '.1.3.6.1.4.1.4329.15.3.5.2.1.0',
+                'ewc',
+                0,
+                'Connected APs'
+            )
+        );
+
+        foreach ($apstats as $index => $entry) {
+            $apStatsMuCounts = $entry['apStatsMuCounts'];
+            $name = $apnames[$index];
+            $sensors[] = new WirelessSensor(
+                'clients',
+                $this->getDeviceId(),
+                "1.3.6.1.4.1.4329.15.3.5.2.2.1.14.$index",
+                'ewc',
+                $index,
+                "Clients ($name)",
+                $apStatsMuCounts
+            );
+        }
+        return $sensors;
+    }
+
+    /**
+     * Discover wireless capacity. Type is capacity.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array Sensors
+     */
+    public function discoverWirelessApCount()
+    {
+        $apactive =  $this->getCacheByIndex('apActiveCount', "HIPATH-WIRELESS-HWC-MIB");
+        return array(
+            new WirelessSensor(
+                'ap-count',
+                $this->getDeviceId(),
+                '.1.3.6.1.4.1.4329.15.3.5.2.1.0',
+                'ewc',
+                0,
+                'Connected APs',
+                $apactive[0]
```

Comment:
You still don't need to fetch it if all devices of this OS support that oid.

Adding max capacity, you do need to fetch that, but yeah the cisciwlc code is probably close to what you want here.